src/buildings/warehouse/warehouse_building.gd

```diff
@@ -0,0 +1,128 @@
+extends Building
+class_name WarehouseBuilding
+
+var specialized_type: EnumAutoload.ResourceType = EnumAutoload.ResourceType.NONE
+
+const SPEC_STORAGE_MULTIPLIER = 4
+const SPEC_PROD_BONUS = 1
+const BASE_PROD_BONUS = 0.2
+
+func _ready():
+	super()
+	data.type = EnumAutoload.BuildingType.STORAGE
+
+
+func apply_upgrades():
+	if EnumAutoload.UpgradeId.CONSTRUCTION_LOGIC_ADV_LOGISTIC in ResourceManager.current_upgrades:
+			get_node("Range").scale = Vector2(1.5, 1.5)
+	# Wait 2 frame to make sure all Area2D changes are setup correctly
+	await get_tree().physics_frame
```

Comment:
Could we maybe replace this with a less general `await`? Something like `Object`'s `property_list_changed` signal? Depends on what exactly we're waiting for here, I'm assuming for the comment an Area2D's shape has been changed and we're waiting for the physics process to update with the new collision data?

Reply:
Yeah, this is for the case buy the increased range upgrade for warehouses. Built warehouse won't be able to increase the it's range Area2D and check for collision fast enough in the same frame so I made it wait a bit.

Reply:
I cant find other solution so we have to keep these 2 await line tho.